opcua/server/address_space.py

```diff
@@ -491,43 +491,75 @@ def empty(self):
 
     def dump(self, path):
         """
-        dump address space as binary to file
+        dump address space as binary to file; note that server must be stopped for this method to work
         """
-        s = shelve.open(path, "n", protocol = pickle.HIGHEST_PROTOCOL)
+        # prepare nodes in address space for being serialized
+        for nodeid in self._nodes.keys():
+            node = self._nodes[nodeid]
+
```

Comment:
iterating though a dict can/should be done with 

    for k, v in mydict.items():
         whaterver